pyproject.toml

```diff
@@ -24,6 +24,7 @@ func-timeout = "^4.3.5"
 assertpy = "^1.1"
 pytest-html = "^3.1.1"
 pluggy = "^0.13.1"
+PyGObject = {version = "^3.38.0", platform = 'linux'}
```

Comment:
Good catch! I have a test on my Ubuntu WSL. It needs to add below package. I will update document. Would you like to fix the github actions also?

`sudo apt install libgirepository1.0-dev libcairo2-dev`

Reply:
Sure, where is that located?

Reply:
in `/Makefile`

Reply:
add one more `gir1.2-secret-1`, but I still get error on my WSL

```log
2021-03-16 23:56:50.185 ERROR LISA.stderr libsecret did not perform properly.
2021-03-16 23:56:50.186 ERROR LISA.stderr * If you encountered error "Remote error from secret service:
2021-03-16 23:56:50.186 ERROR LISA.stderr   org.freedesktop.DBus.Error.ServiceUnknown",
2021-03-16 23:56:50.187 ERROR LISA.stderr   you may need to install gnome-keyring package.
2021-03-16 23:56:50.188 ERROR LISA.stderr * Headless mode (such as in an ssh session) is not supported.
2021-03-16 23:56:50.188 ERROR LISA.stderr Traceback (most recent call last):
2021-03-16 23:56:50.189 ERROR LISA.stderr   File "/home/squirrel/anaconda3/envs/nni/lib/python3.8/site-packages/msal_extensions/libsecret.py", line 126, in trial_run
2021-03-16 23:56:50.190 ERROR LISA.stderr     agent.save(payload)  # It would fail when running inside an SSH session
2021-03-16 23:56:50.190 ERROR LISA.stderr   File "/home/squirrel/anaconda3/envs/nni/lib/python3.8/site-packages/msal_extensions/libsecret.py", line 101, in save
2021-03-16 23:56:50.191 ERROR LISA.stderr     return Secret.password_store_sync(
2021-03-16 23:56:50.192 ERROR LISA.stderr gi.repository.GLib.GError: g-io-error-quark: Cannot autolaunch D-Bus without X11 $DISPLAY (0)
2021-03-16 23:56:50.193 ERROR LISA.stderr libsecret did not perform properly.
2021-03-16 23:56:50.193 ERROR LISA.stderr * If you encountered error "Remote error from secret service:
2021-03-16 23:56:50.193 ERROR LISA.stderr   org.freedesktop.DBus.Error.ServiceUnknown",
2021-03-16 23:56:50.194 ERROR LISA.stderr   you may need to install gnome-keyring package.
2021-03-16 23:56:50.194 ERROR LISA.stderr * Headless mode (such as in an ssh session) is not supported.
```

Reply:
Wait, then has it worked on WSL before already? We should complain to azure-mgmt-* guys: it's very, very silly to depend on pycairo/pygobject for that secret thing--those are GUI packages!

Reply:
I remember it worked before. The python repo is below, we may file an issue to them. We already filed some issues about inconsistent behavior with CLI or PowerShell.
https://github.com/Azure/azure-sdk-for-python

Reply:
Yeah, if it works on systems with a GUI but not on headless one, let's file it.

Reply:
https://docs.github.com/en/actions/using-github-hosted-runners/about-github-hosted-runners --> too bad there is no WSL support on the CI side for us :( I'll not bother with WSL for now, maybe later.